### PR TITLE
1856: Step & Auto Action to check if corporations destinate

### PR DESCRIPTION
**Pull request by ventusignis**
To use `auto_actions` for escrow release I needed an Action that just signified which corporations destinated. To do this I created a general `Engine::Action::Auto` that just stores a string `details`

### CODE REVIEW

File: lib/engine/step/g_1856/escrow.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    module G1856
+      class Escrow < Base
+        # Since *any* corporation could destinated after a given corporation does a tile lay, we need
+        # to check for *all* corporations, and since multiple corporations could destinate at once
+        # we need to be able to support multiple destinating at once
+        def auto_actions(entity)
+          [
+            Engine::Action::Auto.new(
+              entity,
+              details: @game.corporations.select { |c| @game.destinated?(c) }.compact.map(&:id).join('/')
```
Comment (jenf) [inline lib/engine/step/g_1856/escrow.rb:16]: Assuming it works, this would be better as a list

File: lib/engine/game/g_1856.rb [hunk 2]
```diff
@@ -312,13 +312,8 @@ def cert_limit
...
       end
 
-      def destination_check!
-        @corporations.each do |corp|
-          # The capitalization method of unparred corporations is nil
-          next unless corp.capitalization == :escrow
-
-          destinated!(corp) if hexes_connected?(*@destinations[corp.id])
-        end
+      def destinated?(corp)
```
Comment (jenf) [inline lib/engine/game/g_1856.rb:315]: destinated seems weird. 
destination_connected?

File: lib/engine/action/auto.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+require_relative 'base'
+
+# The Auto action type exists for auto_actions
+# This is to be used when the action that is automatically being done
+# is a basic check or otherwise not worth creating an Action class
+# Details is a string.
+module Engine
+  module Action
+    class Auto < Base
+      attr_reader :details
```
Comment (jenf) [inline lib/engine/action/auto.rb:12]: I think I'd prefer this as a specific DestinationConnected with a list of corps

Comment (ventusignis) [reply lib/engine/action/auto.rb]: I renamed it to DestinationConnection as it's vague enough to be reused elsewhere :)

File: lib/engine/game/g_1856.rb (hunk 2 above)
Comment (ventusignis) [reply lib/engine/game/g_1856.rb]: Renamed

File: lib/engine/step/g_1856/escrow.rb (hunk 1 above)
Comment (ventusignis) [reply lib/engine/step/g_1856/escrow.rb]: It's a list now

File: lib/engine/step/g_1856/escrow.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+          [
+            Engine::Action::DestinationConnection.new(
+              entity,
+              corporations: @game.corporations.select { |c| @game.destination_connected?(c) }
+            ),
+          ]
+        end
+
+        def actions(_entity)
+          %w[destination_connection]
```
Comment (tobymao) [inline lib/engine/step/g_1856/escrow.rb:22]: constant

Comment (ventusignis) [reply lib/engine/step/g_1856/escrow.rb]: fixed

**Review by tobymao — APPROVED** [review]

**Pull request merged**